assets/app/view/game/buy_trains.rb

```diff
@@ -68,8 +68,11 @@ def render
         if must_buy_train
 
           children << h('div.margined',
-                        "#{@corporation.name} must buy a train either from The Depot, The Discard"\
-                        "#{other_corp_trains.any? ? ' or other corporations' : ''}")
+                        "#{@corporation.name} must buy a train"\
+                        "#{round.can_buy_train? || !@depot.discarded.empty? ? ' either' : ''}" \
```

Comment:
should it be round.can_buy_train? && @depot.discarded.any?

Reply:
ah i see, because you can always buy from the depot, you're checking if you can buy trains from others or the discard to choose either, 

let's make it round.can_buy_train? || @depot.discarrded.any?

Reply:
@tobymao: yes, the idea was to get the test "XXX must buy a train from The Depot" if it is the only place to buy a train, and to get "XX must buy a train either from The Depot...." if there is more places.
Now I am not native English speaking so I do not know if it would work to remove "either" completly?
Also, would it be possible to have a situation there there a no trains available in The Depot? I know there are variants where the trains can run out, so it might not be 100%.

Another solution to this might be to change the text to 
  **XXX must buy an available train.**

Reply:
yes, in 1836, there was a game when all the diesels were bought.

the either is not necessary,

X must by a train from the depot, the discord, or other corporations is valid

XXX must buy an available train is a good solution and the simplest.